src/core/sync/mutex.d

```diff
@@ -170,8 +171,9 @@ class Mutex :
         else version( Posix )
         {
             int rc = pthread_mutex_unlock( &m_hndl );
+            static immutable syncEx = new SyncError("Unable to unlock mutex");
             if( rc )
-                throw new SyncError( "Unable to unlock mutex" );
+                throw syncEx;
```

Comment:
The exception mechanism doesn't integrate at all with `const`, `immutable` and `shared`, so this is no good.

`onOutOfMemoryError` does it like this:
```d
throw cast(OutOfMemoryError) cast(void*) typeid(OutOfMemoryError).init;
```
But I'm guessing `TypeInfo` isn't TLS either, so it's really almost as bad. Statically allocated exceptions might be an acceptable hack for `Error` exceptions, but it's not a long-term solution unless we change the chaining mechanism.

See [threads such as this one](http://forum.dlang.org/post/stlslhjndgugecvmbowd@forum.dlang.org) for more information.